### PR TITLE
Create Order and Read Order

### LINKED ISSUE #4: Create Order
**As a** customer
**I need** to create a new order  
**So that** I can purchase products from the eCommerce site
   
 ### Details and Assumptions
1. An order consists of multiple order items
2. Each order item represents a product id, its quantity, and its price
3. The order should be associated with a customer id
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given a customer is logged in
 When  they submit a new order with valid product ids, quantities, and customer id
 Then  a new order is created in the system with a unique order id
```

### LINKED ISSUE #23: Read Order (by order id)
**As a** customer and administrator
**I need** to be able to get all the details of an order
**So that** I can decide to edit/cancel (as a customer) / take actions to complete the order (as an administrator)
   
 ### Details and Assumptions
Customer/Administrator needs all information for a given order id
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given order id
 When Read Order API call is made
 Then all details of the order (customer id, items, price, shipping etc.) will be displayed
```

### PR DESCRIPTION
**Pull request by Hyxpillow**
This PR addresses the two issues "Create Order" and "Read Order (by order id)".

1. Added **create_order()** in routes.py.
2. Added **test_create_order()** in test_routes.py
3. Added **read_order()** in routes.py which is responsible for fetching an order by its unique ID
4. Added **test_read_order()** in test_routes.py

resolve #4 
resolve #23

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -95,4 +94,45 @@ def tearDown(self):
...
+        )
+        self.assertEqual(new_order["items"], order.items, "Email does not match")
+
+        # Check that the location header was correct by getting it
+        resp = self.client.get(location, content_type="application/json")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        new_order = resp.get_json()
+        self.assertEqual(new_order["status"], order.status, "Names does not match")
+        self.assertEqual(
+            new_order["customer_name"], order.customer_name, "Address does not match"
```
Comment (anuva98) [inline tests/test_routes.py:125]: The message here should be "customer_name does not match"

File: tests/test_routes.py [hunk 2]
```diff
@@ -95,4 +94,45 @@ def tearDown(self):
...
+        )
+        self.assertEqual(new_order["items"], order.items, "Email does not match")
```
Comment (anuva98) [inline tests/test_routes.py:127]: same as above: "items don't match"

File: tests/test_routes.py [hunk 3]
```diff
@@ -95,4 +94,45 @@ def tearDown(self):
...
+        )
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+
+        # Make sure location header is set
+        location = resp.headers.get("Location", None)
+        self.assertIsNotNone(location)
+
+        # Check the data is correct
+        new_order = resp.get_json()
+        self.assertEqual(new_order["status"], order.status, "Names does not match")
```
Comment (anuva98) [inline tests/test_routes.py:113]: Message: "status doesn't match"

File: tests/test_routes.py [hunk 4]
```diff
@@ -95,4 +94,45 @@ def tearDown(self):
...
+        self.assertEqual(
+            new_order["customer_name"], order.customer_name, "Address does not match"
+        )
+        self.assertEqual(new_order["items"], order.items, "Email does not match")
```
Comment (anuva98) [inline tests/test_routes.py:117]: same as above ^

**Review by anuva98 — CHANGES_REQUESTED** [review]

File: tests/test_routes.py (hunk 4 above)
Comment (Hyxpillow) [reply tests/test_routes.py]: All Done~

**Review by anuva98 — APPROVED** [review]
Thank you. LGTM.

**Review by rheachandok — APPROVED** [review]

**Review by Kirei-Nan — APPROVED** [review]

**Pull request merged**